### ISSUE TITLE
Bug: `snippets` not respecting the `prefix` of the VSCode snippet spec

### ISSUE THREAD

**Issue by chrisgrieser** [antonk52/basics-language-server#4]
Given this snippet in a snippet file:

```json
"foobar": {
  "body": "Hello World",
  "prefix": "barbaz"
},
```

basics_ls only suggests the snippet when typing `foobar`. However, [the vscode snippet spec](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_create-your-own-snippets) specifies that not the entry's key (in the example: `foobar`), but rather the entry's `prefix` value (in the example: `barbaz`) should be used as trigger word:

> **prefix** defines one or more trigger words that display the snippet in IntelliSense. Substring matching is performed on prefixes, so in this case, "fc" could match "for-const".

This is also how all other snippet engines I have used in the past (nvim-snippets, luasnip) do work.

**Comment by antonk52** [comment]
Thanks for the issue, I can look into why this fails to work later today/tomorrow

**Comment by antonk52** [comment]
I've released a new version 1.0.2 which includes the fix and tests for it to make sure this does not happen in the future. This issue should be resolved now.
